app/src/main/java/fr/free/nrw/commons/location/LocationServiceManager.java

```diff
@@ -214,8 +215,12 @@ public void onProviderDisabled(String provider) {
         Timber.d("Provider %s disabled", provider);
     }
 
-    public boolean isProviderEnabled(String provider) {
-        return locationManager.isProviderEnabled(provider);
+    public boolean isNetworkProviderEnabled() {
+        return locationManager.isProviderEnabled(String.valueOf(R.string.network_provider));
```

Comment:
These should be constants or enums

Reply:
I am able to understand what you mean by enums but the constants part is confusing me. I am sorry but can you please explain it to me? Are you referring that I create a `String` variable in `LocationManager` with a `final` attribute like this:

`private final String NETWORK_PROVIDER = "network";`

Reply:
eg `LocationManager.GPS_PROVIDER`

Reply:
Enums or StringDefs would be better, constants like the one you mentioned are fine, if you put them in string files, they will be localized and we don't want that right?

Reply:
Oh! Now I got it. Thank you :)

Reply:
The constants already exist in the framework, so just use those @Glitch101